BuildScript.Nuke/GenerateSbom/SolutionSbomGeneratorBuilder.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership.  rubicon licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.  You may obtain a copy of the
+// License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using Nuke.Common.IO;
+using Nuke.Common.ProjectModel;
+using SbomCleaner.Library;
+
+namespace Remotion.BuildScript.GenerateSbom;
+
+public class SolutionSbomGeneratorBuilder: SbomGeneratorBuilderBase
+{
+  private readonly Solution _solution;
+  private List<string> _packageBlackList = new();
+  private List<string> _projectBlackList = new();
+
+  private AbsolutePath _outputFile;
+  private AbsolutePath _tempDirectory;
+  private AbsolutePath _pathToPackageJson;
+
+  public SolutionSbomGeneratorBuilder (Solution solution, AbsolutePath tempDirectory, AbsolutePath outputFile, string githubUsername, string githubAccessToken)
+  {
+    _solution = solution;
+    _tempDirectory = tempDirectory;
+    _outputFile = outputFile;
+
+    base.WithGithubCredentials(githubUsername, githubAccessToken);
+  }
```

Comment:
I know, builder pattern and all that but is there actually a benefit to using WithGithubCredentials in the base class instead of keeping them in this builder?

And following that, is there a benefit in the base-class vs. interface?

Reply:
Yea, since we dont also have the csproj builder, we can just put them in this directly